Server/src/services/DataSetService.ts

```diff
@@ -345,6 +346,9 @@ export class DataSetService {
             this.requestResponse.message = response as any
             return this.requestResponse
         } catch (error) {
+            if (error instanceof NotFound) {
+                throw new NotFound(error.message)
+            }
             throw new InternalServerError("Something went wrong deleting a saved data set. Try later")

```

Comment:
This should be wrapped. If an error of type NotFound is thrown then this will automatically thrown an InternalServerError. Instead this should be in a if else